src/parser/priest/discipline/modules/spells/Castigation.tsx

```diff
@@ -20,12 +19,12 @@ class Castigation extends Analyzer {
 
   _isCastigationBolt = false;
 
-  constructor(...args) {
-    super(...args);
+  constructor(options: any) {
+    super(options);
     this.active = this.selectedCombatant.hasTalent(SPELLS.CASTIGATION_TALENT.id);
   }
 
-  on_byPlayer_damage(event) {
+  on_byPlayer_damage(event: PenanceDamageEvent) {
```

Comment:
Just a `DamageEvent` (cant assume all events grabbed by this listener are penance events)

The way you can typecheck things like this is with typeguards (refer to previous comment)
```typescript
on_byPlayer_damage(event: PenanceDamageEvent) {
  if(!IsPenanceDamageEvent(event)) {
    return;
  }
  // Everything after that check will automatically cast 'event' to PenanceDamageEvent

  // If you trust your other modules to only add the penanceBoltNumber property to penance events you could technically even remove the event.ability.guid check here now
  if (event.ability.guid !== SPELLS.PENANCE.id || event.penanceBoltNumber !== 3) {
      this._isCastigationBolt = false;
      return;
    }
```
Same with the Heal event listener